maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/TestLocalCompatibilityMojo.java

```diff
@@ -91,43 +91,96 @@ protected ParsedSchema loadSchema(File path, Map<String,
         + "with schema type as %s", path, schemaType));
   }
 
-  public void execute() throws MojoExecutionException {
+  protected ArrayList<File> getFiles(File previousSchemaPath) {
 
-    List<SchemaProvider> providers = MojoUtils.defaultSchemaProviders();
-    Map<String, SchemaProvider> schemaProviders = providers.stream()
-        .collect(Collectors.toMap(SchemaProvider::schemaType, p -> p));
+    ArrayList<File> previousSchemaFiles = new ArrayList<>();
 
-    getLog().debug(String.format("Loading Schema at %s", schemaPath));
-    ParsedSchema schema = loadSchema(schemaPath, schemaProviders);
+    getLog().debug(String.format("Loading File %s", previousSchemaPath));
+    // Add all files inside a directory, inside directories are skipped
+    if (previousSchemaPath.isDirectory()) {
 
-    getLog().debug("Loading Previous Schemas");
-    ArrayList<ParsedSchema> previousSchemas = new ArrayList<>();
-    for (File previousSchemaPath : previousSchemaPaths) {
-      previousSchemas.add(loadSchema(previousSchemaPath, schemaProviders));
+      File[] fileList = previousSchemaPath.listFiles();
+      if (fileList == null) {
+        return previousSchemaFiles;
+      }
+
+      for (File f : fileList) {
+        if (!f.isDirectory()) {
+          previousSchemaFiles.add(f);
+        }
+      }
+
+    } else {
+      previousSchemaFiles.add(previousSchemaPath);
     }
 
-    CompatibilityChecker checker = CompatibilityChecker.checker(compatibilityLevel);
+    return previousSchemaFiles;
+  }
 
-    List<String> errorMessages = checker.isCompatible(schema, previousSchemas);
 
-    if (previousSchemas.size() > 1
+  protected void testSchema(String key, Map<String, SchemaProvider> schemaProviders)
+      throws MojoExecutionException {
+
+    File schemaPath = schemas.get(key);
+
+    if (!previousSchemaPaths.containsKey(key)) {
+      throw new MojoExecutionException(String.format("Previous schemas not found for %s", key));
+    }
+
+    File previousSchemaPath = previousSchemaPaths.get(key);
+    String schemaType = schemaTypes.getOrDefault(key, AvroSchema.TYPE);
+
+    if (!compatibilityLevels.containsKey(key)) {
+      throw new MojoExecutionException(String.format("Compatibility Level not found for %s", key));
+    }
+
+    CompatibilityLevel compatibilityLevel = compatibilityLevels.get(key);
+
+    ArrayList<File> previousSchemaFiles = getFiles(previousSchemaPath);
+
+    if (previousSchemaFiles.size() > 1
         && (compatibilityLevel == CompatibilityLevel.BACKWARD
         || compatibilityLevel == CompatibilityLevel.FORWARD
         || compatibilityLevel == CompatibilityLevel.FULL)) {
 
-      getLog().info(String.format("Checking only with latest Schema at %s",
-          previousSchemaPaths.get(previousSchemaPaths.size() - 1)));
+      throw new MojoExecutionException(String.format("Provide exactly one file for %s check "

```

Comment:
It may be more appropriate to provide reasons for errors rather than suggestions in the error message.